xonsh/jobs.py

```diff
@@ -41,13 +42,16 @@ def wait_for_active_job(signal_to_send=None):
             return
         while obj.returncode is None:
             try:
-                obj.wait(0.01)
+                outs, errs = obj.communicate(timeout=0.01)
             except TimeoutExpired:
                 pass
             except KeyboardInterrupt:
                 obj.kill()
+                outs, errs = obj.communicate()
         if obj.poll() is not None:
             builtins.__xonsh_active_job__ = None
+            obj.stdout = BytesIO(outs)
+            obj.stderr = BytesIO(errs)
```

Comment:
@scopatz ... I don't think I can append to the streams. 
```
$ cmd = ['C:\\Windows\\system32\\cmd.exe', '/C', 'help']
$ proc = Popen(cmd, universal_newlines=False, stdin=0, stdout=PIPE, stderr=PIPE)
$ proc.stdout.writable()
False
```
Is it important? We know the `stderr` and `stdout` streams have been emptied.